CallPlayAudio/src/main/java/com/communication/CallPlayAudio/Ngrok/NgrokService.java

```diff
@@ -0,0 +1,107 @@
+package com.communication.CallPlayAudio.Ngrok;
+
+import net.minidev.json.JSONArray;
+import com.communication.CallPlayAudio.Logger;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Map;
+
+public class NgrokService {
+    /// The NGROK process
+    private Process ngrokProcess;
+    // NgrokConnector connector;
+    private final NgrokConnector connector;
+
+    public NgrokService(String ngrokPath, String authToken) {
+        connector = new NgrokConnector();
+        this.ensureNgrokNotRunning();
+        this.createNgrokProcess(ngrokPath, authToken);
+    }
+
+    /// <summary>
+    /// Ensures that NGROK is not running.
+    /// </summary>
+    private void ensureNgrokNotRunning() {
+        BufferedReader input;
+        String ngrokProcess = "ngrok.exe";
+        String processFilter = "/nh /fi \"Imagename eq " + ngrokProcess + "\"";
+        String tasksCmd = System.getenv("windir") + "/system32/tasklist.exe " + processFilter;
```

Comment:
Windows is not an explicit requirement for your sample (according to the readme file).
Can we make this extensible to other operating systems?

Reply:
will look in later versions